src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -254,12 +260,24 @@ internal void CreateLocalVariable(string name, IValue value)
 
                 if (this.sharedState.CurrentBlock != null)
                 {
-                    // Create the debug info for the local variable declaration
-                    dIBuilder.InsertDeclare(
-                        storage: value.Value,
+                    if (isMutableBinding)
+                    { // variable is represented as a pointer to the value
```

Comment:
nit: It would be good if the comment was in its own line. Same for the other similar comment after the `else` statement.